cmd/influxd/run.go

```diff
@@ -176,6 +176,26 @@ func Run(config *Config, join, version string, logWriter *os.File) (*messaging.B
 				log.Fatalf("failed to start %s Graphite server: %s", c.Protocol, err.Error())
 			}
 		}
+
+		// Start up self-monitoring if enabled.
+		if config.Statistics.Enabled {
+			database := config.Statistics.Database
+			policy := config.Statistics.RetentionPolicy
+
+			// Ensure database exists.
+			if err := s.CreateDatabaseIfNotExists(database); err != nil {
+				log.Fatalf("failed to create database %s for internal statistics: %s", database, err.Error())
+			}
+
+			// Ensure retention policy exists.
+			rp := influxdb.NewRetentionPolicy(policy)
+			if err := s.CreateRetentionPolicyIfNotExists(database, rp); err != nil {
+				log.Fatalf("failed to create retention policy for internal statistics: %s", err.Error())
+			}
+
+			s.StartSelfMonitoring(database, policy, interval)
```

Comment:
Very good -- I just fixed that on my local copy and see you spotted it. Fix coming.